openpype/hosts/maya/plugins/publish/collect_yeti_rig.py

```diff
@@ -31,6 +32,9 @@ def process(self, instance):
 
         input_connections = self.collect_input_connections(instance)
 
+        instance.extend(cmds.sets("input_SET", query=True))
+        instance.extend(get_all_children(cmds.sets("input_SET", query=True)))
```

Comment:
Watch out.

`cmds.sets` returns None if the set is empty.

```python
from maya import cmds
my_set = cmds.sets(empty=True)
print(cmds.sets(my_set, query=True))
# None
```

Reply:
Also, the `get_all_children` function itself does not filter to DAG nodes. So if the set in any way contains non dag nodes, like e.g. another set or dependency node the `get_all_children` function will likely error on `sel.getDagPath()` logic.

You'd need to make sure to pass only dagNode types.

```python
members = cmds.sets("input_SET", query=True) or []
children = get_all_children(cmds.ls(members, type="dagNode", long=True))
instance.extend(members)
instance.extend(children)
```

Reply:
And another thing to watch out for. **If** the instance already contained any of the nodes that are in the `input_SET` then now the instance will have these nodes duplicated (or at least added more than once to the instance).

As such validators will check these nodes twice, and will likely report them more than once too.
It'll be slower + harder to read for the end user.

We might want to ensure uniqueness of what ends up in the instance here?

Reply:
And actually reading in this now - why do we require all children to be collected of the `input_SET`. I recall that by original design that usage of that would not require that at all and wasn't intended like that **at all.** There is something fishy here.

A lot of discussion related to that I think is here: https://github.com/ynput/OpenPype/issues/3404

Reply:
It's best explained in [this comment](https://github.com/ynput/OpenPype/issues/3404#issuecomment-1238491559).

But in practice it means that the nodes in the `input_SET` are nodes that are **inside** your yeti rig.
So say it's `rig_GRP|input_GRP|rig_head_GEO`. When that particular node is found in the `input_SET` then it'll search for incoming connections to that transform and/or shape from any node outside of the rig.

The idea is that you e.g. build you yeti RIG like this:
- You load the referenced model for a character in pipeline, let's say that mesh is at `character/head_GEO`
- You create the yeti rig so that your yeti graph node uses a MESH inside the rig as input. This is `rig_GRP|input_GRP|rig_head_GEO`.
    - This gives the yeti rig full control on that geometry to do additional things on top of whatever input comes in, e.g. deleting faces, pushing faces in/out, subdividing, etc.
- You create a connection between the loaded character's head_GEO and the rig's rig_head_GEO. (e.g. `outMesh` -> `inMesh`).
- You add `rig_GRP|input_GRP|rig_head_GEO` to the `input_SET` because whenever loading the rig you want to re-apply its inputs.
- The yeti rig publisher will then collect the INPUTS found from the rig's head geo `rig_GRP|input_GRP|rig_head_GEO` so that it can later recreate that connection based on its `cbId` so that `{cbId}.output -> rig_head_GEO.input` gets recreated.

Reply:
So these docs are wrong?

https://openpype.io/docs/artist_hosts_maya_yeti#creating-and-publishing-yeti-rig

Reply:
Yes. I'd say the the docs are wrong unless we change the logic of collecting the input set around.

At least currently the code logic doesn't match what the docs describe.

Reply:
Tried giving this workflow a go and something seems off. The publish works well and it produces a `.rigsettings` file with correct IDs and attributes but when loading, this information is taken into account. From the code it looks like IDs of the source and destination nodes need to be same and even then it'll only connect `outMesh` > `inMesh`; https://github.com/ynput/OpenPype/blob/develop/openpype/hosts/maya/plugins/load/load_yeti_rig.py#L65-L77

Reply:
That logic was implemented on OpenPype's side [here](https://github.com/ynput/OpenPype/commit/66544273fd08fc22aedc4267b5716d0dbb8abd23) and admittedly it's logic we (Colorbleed) thus had never used on our end on the intended implementation since it wasn't built to do so.

Instead it should be using the `sourceAttr` and `destAttr` stored in the rigSettings. I'd actually even go as far to by default NOT connect it on load unless someone ticks some "auto-connect" checkbox somewhere. We originally prototyped in our previous pipeline with a 'Connection Manager' which would show the possibilities of connections between A and B by the detected `cbId` entries.

Reply:
> What do you think of using the same workflow as the Xgen implementation, where the "connection manager" is the inventory and the user selects which containers to connect together?

I think that's a solid first step after which we can look into the limitations of that workflow.